src/layer/S3SLSTRLayer.ts

```diff
@@ -49,4 +49,26 @@ export class S3SLSTRLayer extends AbstractSentinelHubV3Layer {
       hasMore: response.data.hasMore,
     };
   }
+
+  public async findDates(
+    bbox: BBox,
+    fromTime: Date,
+    toTime: Date,
+    datasetSpecificParameters: {
+      maxCloudCoverage?: number;
+      orbitDirection: OrbitDirection | null;
+      view: 'NADIR' | 'OBLIQUE';
+    } = { orbitDirection: OrbitDirection.DESCENDING, view: 'NADIR' },
+  ): Promise<Date[]> {
```

Comment:
Elsewhere (in `constructor`s, `findTiles`,...) we change the function signature to include the dataset-specific parameters... I find that approach more intuitive, would you agree? In this case, I would suggest changing the function signature to:
```
public async findDates(
    bbox: BBox,
    fromTime: Date,
    toTime: Date,
    maxCloudCoverage: number | null = null;
    orbitDirection: OrbitDirection | null = null;
    view: 'NADIR' | 'OBLIQUE' = 'NADIR';
): Promise<Date[]> {
```

There's also another thing: some of these parameters (like `maxCloudCoverage`) are now a property of a layer, which makes more sense and simplifies the signature of function. But I guess when merging from master, you'll change that anyway (sorry for the extra work... :-/).

Reply:
I went a different path - I chose the same way the `findTiles` is implemented in EOCloud layers.